uaclient/files/notices.py

```diff
@@ -0,0 +1,206 @@
+
+from uaclient import defaults, event_logger, exceptions, system
+from uaclient.files import UAFile
+
+event = event_logger.get_event_logger()
+NoticeFileDetails = namedtuple(
+    "NoticeFileDetails", ["order_id", "label", "is_permanent"]
+)
+
+
+class Singleton(type):
+    _instances = {}  # type: Dict[Any, Any]
+
+    def __call__(cls, *args, **kwargs):
+        if cls not in cls._instances:
+            cls._instances[cls] = super(Singleton, cls).__call__(
+                *args, **kwargs
+            )
+        return cls._instances[cls]
+
+
+class NoticesDetails(NoticeFileDetails, Enum):
+    REBOOT_REQUIRED = NoticeFileDetails(
+        label="reboot_required", order_id="10", is_permanent=False
+    )
+    ENABLE_REBOOT_REQUIRED = NoticeFileDetails(
+        label="enable_reboot_required", order_id="11", is_permanent=False
+    )
+    REBOOT_SCRIPT_FAILED = NoticeFileDetails(
+        label="reboot_script_failed", order_id="12", is_permanent=True
+    )
+    FIPS_REBOOT_REQUIRED = NoticeFileDetails(
+        label="fips_reboot_required", order_id="20", is_permanent=False
+    )
+    FIPS_SYSTEM_REBOOT_REQUIRED = NoticeFileDetails(
+        label="fips_system_reboot_required", order_id="21", is_permanent=False
+    )
+    FIPS_INSTALL_OUT_OF_DATE = NoticeFileDetails(
+        label="fips_install_out_of_date", order_id="22", is_permanent=True
+    )
+    FIPS_DISABLE_REBOOT_REQUIRED = NoticeFileDetails(
+        label="fips_disable_reboot_required", order_id="23", is_permanent=False
+    )
+    FIPS_PROC_FILE_ERROR = NoticeFileDetails(
+        label="fips_proc_file_error", order_id="24", is_permanent=True
+    )
+    FIPS_MANUAL_DISABLE_URL = NoticeFileDetails(
+        label="fips_manual_disable_url", order_id="25", is_permanent=True
+    )
+    WRONG_FIPS_METAPACKAGE_ON_CLOUD = NoticeFileDetails(
+        label="wrong_fips_metapackage_on_cloud",
+        order_id="25",
+        is_permanent=True,
+    )
+    LIVEPATCH_LTS_REBOOT_REQUIRED = NoticeFileDetails(
+        label="lp_lts_reboot_required", order_id="30", is_permanent=False
+    )
+    CONTRACT_REFRESH_WARNING = NoticeFileDetails(
+        label="contract_refresh_warning", order_id="40", is_permanent=True
+    )
+    DAEMON_AUTO_ATTACH_LOCK_HELD = NoticeFileDetails(
+        label="daemon_auto_attach_lock_held", order_id="50", is_permanent=True
+    )
+    DAEMON_AUTO_ATTACH_FAILED = NoticeFileDetails(
+        label="daemon_auto_attach_failed", order_id="51", is_permanent=True
+    )
+    OPERATION_IN_PROGRESS = NoticeFileDetails(
+        label="operation_in_progress", order_id="60", is_permanent=True
+    )
+
+
+class Notices(metaclass=Singleton):
+    def __init__(
+        self,
+        root_mode: bool = True,
+        override_dir: Optional[str] = None,
+    ):
+        self.directory = override_dir
+        if not override_dir:
+            self.override = False
+        else:
+            self.override = True
+        self.is_root = root_mode
```

Comment:
Ideally I'd like to find a way to test this without needing an override_dir. In theory we can just mock all read/write calls so the dir shouldn't matter and shouldn't need to be overriden.